### DISCUSSION TITLE
Rentable Cache

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#80084] [Q&A]
I am looking for a class that functions like follows:
1.  I "rent" an instance of an item using a key. 
2.  If no instance with that key exists, one is created and returned.  Otherwise, the existing one is returned. 
3.  When I am done with my instance, I release it. 
4.  When an item has no active renters, it is purged from the cache. 

Does a class like this already exist?  If so, what is it called?

**Comment by davidfowl** [comment]
https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.caching.memory.memorycache?view=dotnet-plat-ext-7.0
